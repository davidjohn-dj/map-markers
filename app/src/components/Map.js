import React from "react";
import {
    withGoogleMap,
    GoogleMap,
    Marker,
    withScriptjs
} from "react-google-maps";

const Markers = ({ places }) => {
    return places.map((place, index) => {
        return (
            <Marker key={index} position={{ lat: place.lat, lng: place.lng }} />
        );
    });
};

const Map = ({ places, zoom, center }) => {
    return (
        <GoogleMap defaultZoom={zoom} defaultCenter={center}>
            <Markers places={places} />
        </GoogleMap>
    );
};

class MapWithMarker extends React.Component {
    constructor(props) {
        super(props);
        this.state = { places: this.props.places }; //initialize initial state from props
    }

    componentWillReceiveProps(nextProps) {
        if (this.props.places !== nextProps.places) {
            this.setState({
                places: nextProps.places
            });
        }
        this.setState({
            places: [...nextProps.places]
        });
    }

    render() {
        return (
            <Map
                center={this.props.center}
                zoom={this.props.zoom}
                places={this.state.places}
            />
        );
    }
}

export default withScriptjs(withGoogleMap(MapWithMarker));
