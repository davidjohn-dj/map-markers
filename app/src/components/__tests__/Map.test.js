/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import Map from '../Map';
import { Spin } from 'antd';


describe('Map test', () => {
  it('Map should match snapshot', () => {
    const component = renderer.create(<Map
      googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyBq5bd9aDDbWnOr8hEfyXTeTzHdIO82CL0"
      loadingElement={<Spin size="large" />}
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
