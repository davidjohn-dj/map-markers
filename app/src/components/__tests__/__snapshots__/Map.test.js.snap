// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map test Map should match snapshot 1`] = `
<div
  className="ant-spin ant-spin-lg ant-spin-spinning"
>
  <span
    className="ant-spin-dot ant-spin-dot-spin"
  >
    <i />
    <i />
    <i />
    <i />
  </span>
</div>
`;
